webapp/channels/src/components/edit_category_modal/edit_category_modal.tsx

```diff
@@ -54,12 +58,35 @@ export default class EditCategoryModal extends React.PureComponent<Props, State>
         this.handleClear();
     };
 
+    handleEnterKeyEvent = (e?: React.KeyboardEvent<HTMLDivElement>) => {
+        if (e && e.nativeEvent.isComposing) {
+            return;
+        }
+        this.handleConfirm();
+    };
+
+    componentWillUnmount() {
+        if (this.timeoutId) {
+            clearTimeout(this.timeoutId);
+        }
+    }
+
     handleConfirm = () => {
+        if (this.state.isProcessing) {
+            return;
+        }
+        this.setState({isProcessing: true});
+        this.timeoutId = setTimeout(() => this.setState({isProcessing: false}), ACTION_WAIT_MS);
+
         if (this.props.categoryId) {
             this.props.actions.renameCategory(this.props.categoryId, this.state.categoryName);
         } else {
             this.props.actions.createCategory(this.props.currentTeamId, this.state.categoryName, this.props.channelIdsToAdd);
-            trackEvent('ui', 'ui_sidebar_created_category');
+            trackEvent('ui', 'ui_sidebar_created_category', {
+                currentTeam: this.props.currentTeamId,
+                categoryName: this.state.categoryName,
+                channelds: this.props.channelIdsToAdd,
```

Comment:
What's the reason for tracking these? 

With the category name in particular, I also don't think we can capture that since it feels like we might risk getting confidential information since it's not a random ID like the other two

Reply:
wow, unintended change, was left during debugging. Removing right now.